docs/aws/audit/esmonitoring/rules/es_domain_exposed.mdx

```diff
@@ -23,6 +23,154 @@ HITRUST, AWSWAF, SOC2, NISTCSF, PCIDSS, FedRAMP
+
+4. In the "Modify access policy" section, check the policy document for the "Effect": "Allow" statement and the "Principal": "*" element. If the policy document contains these elements, the selected Elasticsearch domain is publicly accessible, which means it's exposed to the internet.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can run any AWS CLI commands, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided in the AWS CLI User Guide.
+
+2. List all Elasticsearch domains: Use the following AWS CLI command to list all the Elasticsearch domains in your AWS account:
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all Elasticsearch domain names.
+
+3. Describe the Elasticsearch domain: For each domain name returned in the previous step, use the following AWS CLI command to get detailed information about the domain:
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name <your-domain-name>
+   ```
+   Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return a JSON object that contains detailed information about the domain.
+
+4. Check the access policy: In the JSON object returned in the previous step, look for the `AccessPolicies` field. This field contains the access policy for the Elasticsearch domain. If the access policy allows public access (i.e., access from '0.0.0.0/0' or '::/0'), then the Elasticsearch domain is publicly exposed.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. 
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. Create a session using your AWS credentials. You can configure your credentials in the AWS credentials file located by default at `~/.aws/credentials`.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+
+3. Create an Elasticsearch client:
+
+```python
+es_client = session.client('es')
+```
+
+4. List all Elasticsearch domains and check their access policies:
+
+```python
+def check_es_domains_public_access():
+    try:
+        response = es_client.list_domain_names()
+        for domain in response['DomainNames']:
+            domain_name = domain['DomainName']
+            domain_config = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+            access_policies = domain_config['DomainStatus']['AccessPolicies']
+            if '"Effect":"Allow","Principal":{"AWS":"*"}' in access_policies:
+                print(f"Elasticsearch domain {domain_name} is publicly accessible.")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>